src/typescript/azure-openapi-validator/rules/DescriptionAvoidsGerunds.ts

```diff
@@ -0,0 +1,36 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+import { MergeStates, OpenApiTypes, rules } from '../rule';
+
+export const DescriptionAvoidsGerunds: string = "DescriptionAvoidsGerunds";
+
+rules.push({
+  id: "D4004",
+  name: DescriptionAvoidsGerunds,
+  severity: "warning",
+  category: "DocViolation",
+  mergeState: MergeStates.composed,
+  openapiType: OpenApiTypes.doc,
+  appliesTo_JsonQuery: "$..description",
+  run: function* (doc, node, path) {
+    const msg: string = "Description contains possible gerunds.";
+    if (typeof (node) !== 'string') {
+      return;
+    }
+
+    const nodeValue: string = <string>node;
+
+    // Known good words which are not gerunds, but do end in -ing
+    const goodWords = ["string"];
+
+    var gerunds = nodeValue.match(/\b\w+ing\b/g);
```

Comment:
Are we assuming that description will be exactly one word here?

Reply:
No; this matches individual words within the string (`\b` indicates a word boundary.) While this will also detect terms other than gerunds such as present participle verbs, it's unfortunately the best check that can be done without extensive syntax analysis (which is an option that may be added if an excessive number of present participles are detected - though these are ALSO very bad for machine translation.)

Reply:
I'm happy with the regex.